### PR TITLE
terraform-lsp 0.0.12 (new formula)

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/terraform-lsp.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      -s -w
+      -X main.Version=#{version}
+      -X main.GitCommit=#{Utils.git_head}
+      -X main.Date=#{time.iso8601}
+    ]
+
+    system "go", "build", *std_go_args(ldflags: ldflags)
+  end
+
+  test do
```
Comment (carlocab) [inline Formula/terraform-lsp.rb:23]: Can we test the version here too?

Comment (chenrui333) [reply Formula/terraform-lsp.rb:23]: Sure

**Review by nandahkrishna — APPROVED** [review]

File: Formula/terraform-lsp.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    end
+    sleep 2
+
+    begin
+      tcp_socket = TCPSocket.new("localhost", port)
+      tcp_socket.puts <<~EOF
+        Content-Length: 59
+
+        {"jsonrpc":"2.0","method":"initialize","params":{},"id":1}
+      EOF
```
Comment (SMillerDev) [inline Formula/terraform-lsp.rb:37]: Couldn't this be done with cURL?

Comment (chenrui333) [reply Formula/terraform-lsp.rb:37]: This is the similar same test adopted from the `terraform-ls` formula, let me see if I can use `curl` to do it.

Comment (chenrui333) [reply Formula/terraform-lsp.rb:37]: I could not get the curl test working

Comment (chenrui333) [reply Formula/terraform-lsp.rb:37]: I think we can merge the PR and address the test later together with `terraform-ls` one, @SMillerDev what do you think?

**Review by SMillerDev — APPROVED** [review]

**Comment by chenrui333** [comment]
Thanks folks, merging this now.

**Pull request closed without merging**